### DISCUSSION TITLE
Sequence of running addClusters() and addUMAP() functions

### DISCUSSION THREAD

**Discussion by sylestiel** [GreenleafLab/ArchR#1767] [General]
Hi,
 
 I noticed that calling clusters function addClusters() precedes embedding function addUMAP() in the full manual while the order is reversed in the 10X Multiome vignette. Is there any particular reason for it.

Secondly, after running addHarmony() do we run addUMAP with Harmony has the reduced dimension?

Thank you!

**Comment by rcorces** [comment]
> I noticed that calling clusters function addClusters() precedes embedding function addUMAP() in the full manual while the order is reversed in the 10X Multiome vignette. Is there any particular reason for it.

Clusters are independent of the embedding and only rely on the LSI reduced dimensions space. The UMAP is just a way to visualize that LSI dim reduction. So the addClusters() and addUMAP do not affect each other.

> Secondly, after running addHarmony() do we run addUMAP with Harmony has the reduced dimension?

Yes - if you want to look at cells on a UMAP after running Harmony, you absolutely need to run `addUMAP()` using the harmony reduced dims.

**Reply by hira-anis** [reply]
Hey Ryan, 

A follow up question on this,

Working on multiome data, AddIterativeLSI (RNA,ATAC) > AddCombinedDims > RunHarmony (on all three) followed by AddClusters and AddUMAP. Does it sound like the right order? I am facing some issues in adding Clusters on LSI_combined using reduceddims from Harmony for LSI combined.

**Reply by hira-anis** [reply]
https://github.com/GreenleafLab/ArchR/discussions/2068

this is the error.
